### DISCUSSION TITLE
How to remove the __INITIAL_STATE__ variable

### DISCUSSION THREAD

**Discussion by yahao87** [quasarframework/quasar#12999] [CLI - SSR mode] [answered]
I'm using Quasar v1.
It works only as a PWA in the user's browser. Basically, we have a lot of data to store in common, so we're caching this data with localstorage.
And using SSR, it only responds to search engine bots.
Therefore, I do not need the __INITIAL_STATE__ variable, and for optimization, the store must be managed as a singleton in the SSR server.
But I can't see any way to remove the __INITIAL_STATE__ variable.
Too heavy data is being included in every SSR response.
Is there any way to remove the __INITIAL_STATE__ variable?

**Comment by MilosPaunovic** [comment]
[https://github.com/quasarframework/quasar/issues/5903](https://github.com/quasarframework/quasar/issues/5903)

[https://quasar.dev/quasar-cli-webpack/developing-ssr/configuring-ssr#manually-triggering-store-hydration](https://quasar.dev/quasar-cli-webpack/developing-ssr/configuring-ssr#manually-triggering-store-hydration)

```js
ssr: {
  manualStoreHydration: true/false,
    // Manually hydrate the store.
    // This is detailed in a subsection below
}
```

**Reply by yahao87** [reply]
What you've found is how to do the hydration manually.
What I'm looking for is a way to not export the store at all.

**Reply by yahao87** [reply]
I need a way to resolve responses that are over 1 megabyte(gziped) to every ssr page.
Help me.... T.T

**Comment by yusufkandemir** [comment]
If you are sure that you don't need state hydration in any way, you can do the following in `store(s)/index.js` or in a boot file.
```js
// /src/store(s)/index.js
// ...

export default store(function ({ ssrContext }) {
  // ...

  if (process.env.SERVER && ssrContext) {
    ssrContext.onRendered(() => {
      // Disable
      ssrContext.state = undefined
    })
  }
  
  // ...
})
```
See [`ssrContext` docs](https://quasar.dev/quasar-cli-vite/developing-ssr/ssr-context#anatomy-of-ssrcontext) for explanation about `state` and `onRendered`

**Reply by yahao87** [reply]
thanks.
I tried to find it in v1 somehow with the clue you gave me.
But in v1 I can't find the rendered function.
Is there a similar feature in v1?

**Comment by yahao87** [comment]
@yusufkandemir 
Now the SSR response is close to 5MB.
I can't update vue major version.
Should we give up the quasar SSR?
Help me make a decision.

**Reply by MilosPaunovic** [reply]
If you want to update Vue major version to 3.x.x, you have to switch to `Quasar@2.x.x` - Quasar 1 utilizes `Vue@2.x.x`.

**Reply by yahao87** [reply]
> If you want to update Vue major version to 3.x.x, you have to switch to [Quasar@2.x.x](mailto:Quasar@2.x.x) - Quasar 1 utilizes [Vue@2.x.x](mailto:Vue@2.x.x).

I'm just saying I should use Quasar v1. I don't want to upgrade vue version.
Because the project size is large, it takes a lot of time to change it.
I just need to check.
Is there any solution.

**Comment by rstoenescu** [comment, marked as answer]
Going to implement this for Quasar v2 then backport to v1 also.
Please follow the progress in https://github.com/quasarframework/quasar/issues/13113

**Reply by rstoenescu** [reply]
@yahao87 
The functionality that you want will be available through "@quasar/app" v2.3. In your quasar.conf.js file, add this:

```js
// quasar.conf.js
ssr: {
  manualStoreSsrContextInjection: true
}
```

This won't add add the store state to the ssrContext and so it won't serialize and inject it into the server-side output.

I've scheduled multiple Quasar packages releases for tomorrow (April 8th) and this is included.

**Reply by yahao87** [reply]
wow.
I was thinking of another alternative.
I never thought you could add features so quickly.
Thank you so much.
I love quasar
Thanks again

**Reply by rstoenescu** [reply]
And I've just released the new stuff.
Donations from your company are appreciated to get the wheels going ;) https://donate.quasar.dev
